File: python/examples/lux3d_part_split_list.py
#!/usr/bin/env python3
import sys

from manycore.aholo_sdk_core import AholoClientConfig
from manycore.aholo_sdk_lux3d import create_lux3d_client

if len(sys.argv) < 2:
    raise SystemExit("Usage: AHOLO_API_KEY=xxx python lux3d_part_split_list.py <glb-url>")

lux3d = create_lux3d_client(AholoClientConfig(region="cn"))
task_id = lux3d.part_split.create(glb_url=sys.argv[1])
print("part_split task_id=", task_id)
print("recent tasks=", lux3d.tasks.list(page=1, page_size=20))

File: java/examples/src/main/java/com/manycoreapis/sdk/examples/Lux3dPartSplitList.java
package com.manycoreapis.sdk.examples;

import com.manycoreapis.sdk.core.AholoClientConfig;
import com.manycoreapis.sdk.lux3d.Lux3dClient;
import com.manycoreapis.sdk.lux3d.model.PartSplitCreateParams;
import com.manycoreapis.sdk.lux3d.model.TaskListParams;

public final class Lux3dPartSplitList {
    private Lux3dPartSplitList() {}

    public static void main(String[] args) {
        if (args.length < 1) throw new IllegalArgumentException("Usage: Lux3dPartSplitList <glb-url>");
        Lux3dClient lux3d = Lux3dClient.create(AholoClientConfig.ofRegion("cn"));
        long taskId = lux3d.partSplit().create(PartSplitCreateParams.builder().glbUrl(args[0]).build());
        System.out.println("partSplit taskId=" + taskId);
        System.out.println("recent total=" + lux3d.tasks().list(
                TaskListParams.builder().page(1).pageSize(20).build()).total());
    }
}
